### PR TITLE
docs: Sort __all__ list guideline instruction

### LINKED ISSUE #370: Sorting of exceptions in `src/safeds/exceptions/__init.py__` not enforced by linter
### Is your feature request related to a problem?

The exceptions in the `__all__` list in `src/safeds/exceptions/__init__.py` are currently (1752feb06bb5b0f041310956eca2ee5ed08dd999 on `main`) not sorted. Apparently, this is not enforced by the linter.

With the list not sorted, people will add new exceptions generally to the end of the list, producing unnecessary merge conflicts when different exceptions are added on different branches.  
With sorting enforced, merge conflicts would only happen when the new exceptions would come right after another in the alphabet (and thus would have to be inserted at the same position in the list).  
These merge conflicts are always trivial to resolve by hand, but it's still better to have git do that kind of work for us.

### Desired solution

Make the linter enforce that the list is sorted.

### PR DESCRIPTION
**Pull request by PhilipGutberlet**
Closes #370

### Summary of Changes

Sorted the entries in `__all__` list in `__init__.py` files and added an instruction to the guidelines.

### CODE REVIEW

**Comment by zzril** [comment]
Just on a side note: For updating the guidelines, this will already be addressed by another PR, see [this](https://github.com/Safe-DS/Stdlib/issues/370#issuecomment-1624257780) comment, so it was not really necessary to change it.

File: docs/development/guidelines.md [hunk 1]
```diff
@@ -205,6 +205,29 @@ The user should not have to deal with exceptions that are defined in the wrapper
         return pd.read_csv(path) # May raise a pd.ParserError
     ```
 
+### Sort the exceptions in the `__all__` list alphabetically
+
+The exceptions in the `__all__` list in `src/safeds/exceptions/__init__.py` should be sorted alphabetically. This makes it easier for users to find the exception they are looking for.
```
Comment (zzril) [inline docs/development/guidelines.md:210]:
```suggestion
The exceptions in the `__all__` list in `src/safeds/exceptions/__init__.py` should be sorted alphabetically. This helps to reduce the likelihood of merge conflicts when new exceptions are introduced on different branches.
```

Comment (lars-reimann) [inline docs/development/guidelines.md:210]:
```suggestion
### Sort the entries in the `__all__` list alphabetically

The entires in the `__all__` list in `__init__.py` files should be sorted alphabetically. This makes it easier for users to find the exception they are looking for.
```

Comment (PhilipGutberlet) [reply docs/development/guidelines.md]: applied

**Review by robmeth — APPROVED** [review]

**Review by patrikguempel — DISMISSED** [review]

File: docs/development/guidelines.md [hunk 2]
```diff
@@ -205,6 +205,27 @@ The user should not have to deal with exceptions that are defined in the wrapper
         return pd.read_csv(path) # May raise a pd.ParserError
     ```
 
+### Sort entries in `__all__` lists alphabetically
+The entries in the `__all__` list in `__init__.py` files should be sorted alphabetically. This helps reduce the likelihood of merge conflicts when new entries are introduced on different branches.
+!!! success "**DO** (library code):"
+
+    ```py
+    __all__ =[
```
Comment (lars-reimann) [inline docs/development/guidelines.md:213]:
```suggestion
    __all__ = [
```

File: docs/development/guidelines.md [hunk 3]
```diff
@@ -205,6 +205,27 @@ The user should not have to deal with exceptions that are defined in the wrapper
...
+        "ColumnSizeError",
+        "DuplicateColumnNameError",
+        "MissingValuesColumnError"
+    ]
+    ```
+!!! failure "**DON'T** (library code):"
+
+    ```py
+    __all__ =[
```
Comment (lars-reimann) [inline docs/development/guidelines.md:222]:
```suggestion
    __all__ = [
```

**Review by lars-reimann — CHANGES_REQUESTED** [review]

**Review by lars-reimann — APPROVED** [review]

**Pull request merged**